lib/active_merchant/billing/gateways/securion_pay.rb

```diff
@@ -289,6 +313,10 @@ def api_request(endpoint, parameters = nil, options = {}, method = nil)
         response
       end
 
+      def shift4_v2?
+        @options[:url_override].to_s == 'shift4_v2'
```

Comment:
In this case we need only do override the URL, because `securion_pay` use the same type of credentials (secret_key), that's way we don't need a new mode for shift4.